events: Add `is_redacted` accessor to event enums

zecakeh opened the pull request:
Allows to know if an event is redacted without calling `original_content()`, which clones the content.

cc @bnjbvr

bnjbvr reviewed (commented):
Nice, thanks! Not sure what's going on with the tests getting moved or removed, though?

zecakeh commented:
I removed 2 tests that were duplicates. Git complicated the diff a little because I also added the check for `is_redacted`.

Edit: I should probably move the removal to a separate commit.

zecakeh committed:
events: Add is_redacted accessor to event enums

Allows to know if an event is redacted without calling `original_content()`,
which clones the content.

zecakeh committed:
events: Remove test duplicates

zecakeh commented:
I cleaned up the history to clarify what happened with the tests.
